bitnami/postgresql-ha/templates/postgresql/statefulset.yaml

```diff
@@ -122,7 +122,7 @@ spec:
               value: {{ .Values.persistence.mountPath | quote }}
             - name: PGDATA
               value: {{ printf "%s/%s" .Values.persistence.mountPath "data" | quote }}
-            {{- if and .Values.postgresql.postgresPassword (not (eq .Values.postgresql.username "postgres")) }}
+            {{- if or .Values.postgresql.existingSecret (and .Values.postgresql.postgresPassword (not (eq .Values.postgresql.username "postgres"))) }}
```

Comment:
This assumes that you're using a user different from "postgres" when an existing secret is used to store the credentials.